### PR TITLE
allow views of timestamps in constructors

### PR DESCRIPTION
**Pull request by JeffreySarnoff**
allows TimeArrays to be constructed from view_s of another TimeArray's timestamps
just as TimeArrays can be constructed from view_s of another TimeArray's values

### CODE REVIEW

**Comment by ararslan** [comment]
Nice, thanks! Could you add a test for the new functionality?

**Comment by JeffreySarnoff** [comment]
I have not used that style of test --  the old way the test would be e.g.

```julia
    items = 101:121
    cols = 1:size(AAPL.values)[2]
    tstamps = view(AAPL.timestamp, items)
    tvalues = view(AAPL.values, items, cols)
    APPL2 = TimeArray(tstamps, tvalues, AAPL.colnames, AAPL.meta)
    
    @test AAPL[101:121] == AAPL2[1:end]
```

File: test/timearray.jl [hunk 1]
```diff
@@ -55,6 +55,16 @@ facts("type constructors enforce invariants") do
...
 end
 
+facts("type constructors allow views") do    
+    items = 101:121
+    cols = 1:size(AAPL.values)[2]
+    tstamps = view(AAPL.timestamp, items)
+    tvalues = view(AAPL.values, items, cols)
+    APPL2 = TimeArray(tstamps, tvalues, AAPL.colnames, AAPL.meta)
+    
+    @test AAPL[101:121] == AAPL2[1:end]
```
Comment (iblis17) [inline test/timearray.jl]: there is a macro `@fact` from FactCheck...

**Comment by ararslan** [comment]
Yes, as @iblis17 points out, this repository unfortunately still uses the outdated FactCheck package for its testing rather than the unit testing functionality provided in Base. To turn a `@test` into a `@fact`, you do `@fact a --> b` rather than `@test a == b`.

**Comment by JeffreySarnoff** [comment]
I do not see the problem -- it works for me with cut and paste

File: test/timearray.jl [hunk 2]
```diff
@@ -55,6 +55,30 @@ facts("type constructors enforce invariants") do
...
+    
+    source_rows = 101:121
+    source_cols = 1:size(AAPL.values)[2]    
+    
+    AAPL1 = TimeArray(AAPL.timestamp[source_rows], AAPL.values[source_rows, source_cols], AAPL.colnames, AAPL.meta)
+    
+    tstamps = view(AAPL.timestamp, source_rows)
+    tvalues = view(AAPL.values, source_rows, source_cols)
+    
+    APPL2 = TimeArray(tstamps, tvalues, AAPL.colnames, AAPL.meta)
```
Comment (iblis17) [inline test/timearray.jl]: ha, I got it. `APPL2` is typo

Comment (iblis17) [inline test/timearray.jl]: It should be `AAPL2`

File: test/timearray.jl [hunk 3]
```diff
@@ -55,6 +55,30 @@ facts("type constructors enforce invariants") do
     end
 end
 
+facts("type constructors allow views") do
+    
+    source_rows = 101:121
+    source_cols = 1:size(AAPL.values)[2]
```
Comment (iblis17) [inline test/timearray.jl]: please purge the trailing whitespace :)

File: test/timearray.jl (hunk 2 above)
Comment (JeffreySarnoff) [inline test/timearray.jl]: :coffee: very good noticing   :grapes:

**Comment by ararslan** [comment]
Looks great, thanks for the contribution!

**Comment by milktrader** [comment]
Should we move away from FactCheck? I guess this is probably an important issue.

**Comment by JeffreySarnoff** [comment]
yes

**Comment by ararslan** [comment]
Yes, definitely. I believe it's no longer actively maintained, plus the Base unit testing framework is fantastic.

**Comment by iblis17** [comment]
@ararslan Here is the blocker I encountered: https://github.com/JuliaQuant/MarketTechnicals.jl/issues/70#issuecomment-298226219

Our testing relies on tons of `roughly`, but the sweet macro is only available in 0.6.

**Comment by JeffreySarnoff** [comment]
roughly is defined using isapprox
```julia

# approx/roughly: Comparing numbers approximately
roughly(x::Number, atol) = (y::Number) -> isapprox(y, x, atol=atol)

roughly(x::Number; kvtols...) = (y::Number) -> isapprox(y, x; kvtols...)

roughly(A::AbstractArray, atol) = (B::AbstractArray) -> begin
    size(A) != size(B) && return false
    return isapprox(A, B, atol=atol)
end

roughly(A::AbstractArray; kvtols...) = (B::AbstractArray) -> begin
    size(A) != size(B) && return false
    return isapprox(A, B; kvtols...)
end
```

**Pull request merged**